src/core/thread/dua_manager.cpp

```diff
@@ -342,9 +342,18 @@ void DuaManager::HandleBackboneRouterPrimaryUpdate(BackboneRouter::Leader::State
                                                    const BackboneRouter::Config &aConfig)
 {
     OT_UNUSED_VARIABLE(aConfig);
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    Mle::MleRouter &mle = Get<Mle::MleRouter>();
```

Comment:
Use `Get<Mle::Mle>()` directly below? (to avoid additional `#if OPENTHREAD_CONFIG_DUA_ENABLE` check). The `Get<Type>()` methods are inlined.

Reply:
Agree, will do it.

Reply:
Please note that I tried to keep the original logic: e.g.: in [PerformNextRegistation](https://github.com/openthread/openthread/blob/5503071cb20c785a794701b5da9a16f1ec375cb4/src/core/thread/dua_manager.cpp#L423) when a new DUA is registered, the same logic is applied:

`    // Only send DUA.req when necessary
#if OPENTHREAD_CONFIG_DUA_ENABLE
#if OPENTHREAD_FTD
    VerifyOrExit(mle.IsRouterOrLeader() || !mle.IsExpectedToBecomeRouterSoon(), error = kErrorInvalidState);
#endif
    VerifyOrExit(mle.IsFullThreadDevice() || mle.GetParent().IsThreadVersion1p1(), error = kErrorInvalidState);
#endif // OPENTHREAD_CONFIG_DUA_ENABLE`

Also, as far as I know, only an MTD that is attached to an 1.1 parent must send a DUA reg directly to the PBBR. Maybe @librasungirl can advise.